src/openrct2/platform/posix.c

```diff
@@ -844,6 +845,36 @@ utf8* platform_get_username() {
     }
 }
 
+bool platform_get_steam_path(utf8 *outPath, size_t outSize)
+{
+#if defined(__unix__)
+    char *steamRoot = getenv("STEAMROOT");
+    if (steamRoot != NULL)
+    {
+        safe_strcpy(outPath, steamRoot, outSize);
+        safe_strcat_path(outPath, "steamapps/common", outSize);
+        return true;
+    }
+    else
+    {
+        if (platform_directory_exists("~/.local/share/Steam/steamapps/common"))
```

Comment:
`~` is expanded by the shell to the user home path, but this only works in the shell and not in C.
In addition, you should not concatinate the `.local/share` bit either. There is an environment variable for that: `XDG_DATA_HOME`. You can look for the similar working XDG_CONFIG_HOME in the code base to see how to obtain its value and how to use it.

Also, as far as I know, Steam only uses the first path (the one in title case). (Correct me if I'm wrong.)

Reply:
I've heard many conflicting opinions on if ~/.steam/steam really exists but some of the people on Stack Overflow say it does.